docs/cpp_docs.rst

```diff
@@ -26,94 +45,540 @@ The beginning of both header and source code files should include
+4. :ref:`cpp-doxygen-tparameters` (if applicable; for classes, methods, and functions)
+5. :ref:`cpp-doxygen-parameters` (if applicable; for methods and functions)
+6. :ref:`cpp-doxygen-returns` (if applicable; for methods and functions)
+7. :ref:`cpp-doxygen-throws` (if applicable; for methods and functions)
+8. :ref:`cpp-doxygen-exceptsafe` (recommended; for methods and functions)
+9. :ref:`cpp-doxygen-related` (if applicable; for functions)
+10. :ref:`cpp-doxygen-initializer` (if applicable; for constants)
+11. :ref:`cpp-doxygen-see-also` (optional)
+12. :ref:`cpp-doxygen-notes` (optional)
+13. :ref:`cpp-doxygen-references` (optional)
+14. :ref:`cpp-doxygen-examples` (optional)
+
+For summaries of how these sections are composed in specific contexts, see:
+
+- :ref:`cpp-doxygen-package-definition`
+- :ref:`cpp-doxygen-class-structure`
+- :ref:`cpp-doxygen-enum-structure`
+- :ref:`cpp-doxygen-method-function-structure`
+- :ref:`cpp-doxygen-attribute-constants-structure`
+
+.. _cpp-doxygen-short-summary:
+
+Short Summary
+-------------
+
+A one-line summary that does not use variable names or the function's name. This summary will appear in lists of class/namespace members.
+
+.. code-block:: cpp
+
+   /// Sum two numbers.
+   double add(double a, double b);
+
+For functions and methods, the summary should be written in the imperative voice (i.e., as a command that the API consumer is giving). Getters and other methods that are more naturally described as values rather than actions may ignore this rule.
+
+.. _cpp-doxygen-deprecation:
+
+Deprecation Warning
+-------------------
+
+A ``@deprecated`` tag (where applicable) to warn users that the component is deprecated. The text that follows the tag should include:
+
+1. In what stack version the object was deprecated, and when it will be removed.
+2. Reason for deprecation if this is useful information (object is superseded, duplicates functionality found elsewhere, etc.).
+3. New recommended way of obtaining the same functionality.
+
+.. code-block:: cpp
+
+   /**
+    * @deprecated Deprecated in 11_0. `ndobj_old` will be removed in 12_0; it
+    *     is replaced by `ndobj_new` because the latter works also with array
+    *     subclasses.
+    */
+
+.. _cpp-doxygen-extended-summary:
+
+Extended Summary
+----------------
+
+A few sentences giving an extended description.
+This section should be used to clarify *functionality*, not to discuss implementation detail or background theory, which should rather be explored in the :ref:`cpp-doxygen-notes` section below.
+You may refer to the parameters and the function name, but parameter descriptions still belong in the :ref:`cpp-doxygen-parameters` section unless they are very lengthy.
+
+This section may include mathematical equations to describe the behavior of a class or method, but be sure to put math that pertains only to the implementation in the :ref:`cpp-doxygen-notes` section rather than the extended description.
```

Comment:
Math is described in "Notes" in the Python guide rather than in "Extended Summary".  Is the thinking that it's more likely to be used for behavior and not implementation in C++ than in Python?